src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -210,6 +212,13 @@ public override void OnEventWritten(string name, object? payload)
                         {
                             if (this.exceptionFetcher.TryFetch(payload, out Exception? exception) && exception != null)
                             {
+                                activity.AddTag(SemanticConventions.AttributeErrorType, exception.GetType().FullName);
```

Comment:
This is not very useful because `error.type` will always equal `Microsoft.Data.SqlClient.SqlException`.

Per [the conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/mssql.md), this is not necessarily wrong, but it's not ideal:

> The error.type SHOULD match the db.response.status_code returned by the database or the client library, or the canonical name of exception that occurred. When using canonical exception type name, instrumentation SHOULD do the best effort to report the most relevant type. For example, if the original exception is wrapped into a generic one, the original exception SHOULD be preferred. Instrumentations SHOULD document how error.type is populated.

Doing something like the following would be more useful and would align with the response status code.

```csharp
activity.AddTag(SemanticConventions.AttributeErrorType, exception.Message);
```

However exception message can have variable text and could cause error.type to be high cardinality. For example:

```sql         
SELECT ColumnThatDoesNotExist FROM MyTable
```

This query results in [error code 207](https://learn.microsoft.com/sql/relational-databases/errors-events/mssqlserver-207-database-engine-error?view=sql-server-ver16) and message:

> Invalid column name 'ColumnThatDoesNotExist'

It's tempting to do something like the following to replace anything in single-quotes with something.

```csharp
Regex.Replace(exception.Message, "'[^']*'", "'%'"));
```

But, glancing over the list of [error codes](https://learn.microsoft.com/sql/relational-databases/errors-events/database-engine-events-and-errors?view=sql-server-ver16), I'm not sure such a simple replacement will suffice.

Alternatively, we could maintain a dictionary of codes to messages... there are a lot.

Reply:
I believe there is a common practice to set message on the span description.

It's ok if for .NET it's uncommon to subclass exceptions and that `error.type` won't be super-interesting. We still have status code.